src/components/QueryEditor.tsx

```diff
@@ -1,29 +1,75 @@
-import React from 'react';
-import { Button, QueryField } from '@grafana/ui';
+import React, { useMemo } from 'react';
+import { Button, QueryField , Select } from '@grafana/ui';
 import { QueryEditorProps } from '@grafana/data';
 import { DataSource } from '../datasource';
 import { SumoQuery } from '../types/metricsApi.types';
-
+import { SumoQueryType } from '../types/constants';
+ 
 type Props = QueryEditorProps<DataSource, SumoQuery>;
 
-export function QueryEditor({ query, onChange, onRunQuery }: Props) {
+const selectOptions = [
+  {
+    value : SumoQueryType.Logs,
+    label : 'Logs'
+  },
+  {
+    value : SumoQueryType.Metrics,
+    label : 'Metrics'
+  }
+]
+
+export function QueryEditor(props: Props) {
+
+  const { queries , query , onChange , onRunQuery} = props
+
   const onQueryTextChange = React.useCallback((changedQueryText: string) => {
-    onChange({ ...query, queryText: changedQueryText });
-  }, [query]);
+    onChange({ ...query, queryText: changedQueryText  });
+  }, [query , onChange]);
+
+  const { queryText , type , refId  } = query;
+
+  const onTypeChangeHandler = React.useCallback((selectedObj : { value?: SumoQueryType , label? : string })=>{
+    onChange({ ...query, type : selectedObj.value });
+  } , [query , onChange])
+
+
+  const isEditorDisabled = useMemo(()=>{
+    if(props.query.type === SumoQueryType.Logs){
+      return false;
+    }
+    // disable editor is there is any log query present
```

Comment:
why we want to disable logs query editing?

Reply:
Unlike metrics, Search jobs api only support only one query. For Single search we fire multiple search jobs Apis for eg. Create session, polling status, fetching records and delete session. This will be complicated if we support multiple logs query capability. So in order to make thinks simple we disable rest of the query editors in grafana if there is any logs query present in targets array. Please see video for clarification